annotators/custom_entity_linking/server.py

```diff
@@ -0,0 +1,142 @@
+        if len(hist_utt) > 1:
+            prev_utt = hist_utt[-2]
+            if prev_utt[-1] not in {".", "!", "?"}:
+                prev_utt = f"{prev_utt}."
+            opt_context_batch.append([prev_utt, last_utt])
+        else:
+            opt_context_batch.append([last_utt])
+
+    entity_info_batch = [[{}] for _ in entity_substr_batch]
+    try:
+        (
+            entity_substr_batch,
+            entity_ids_batch,
+            conf_batch,
+            entity_id_tags_batch,
+        ) = el(user_ids, entity_substr_batch, entity_tags_batch, opt_context_batch)
+        entity_info_batch = []
+        for (entity_substr_list, entity_ids_list, conf_list, entity_id_tags_list, prex_info, context) in zip(
+            entity_substr_batch, entity_ids_batch, conf_batch, entity_id_tags_batch, prex_info_batch, opt_context_batch
+        ):
+            if context:
+                context = " ".join(context)
+            else:
+                context = ""
+            entity_info_list = []
+            triplet = {}
+            if isinstance(prex_info, list) and prex_info:
+                prex_info = prex_info[0]
+            if prex_info:
+                triplet = prex_info.get("triplet", {})
+            rel = ""
+            if "relation" in triplet:
+                rel = triplet["relation"]
+            elif "property" in triplet:
+                rel = triplet["property"]
+            for entity_substr, entity_ids, confs, entity_id_tags in zip(
+                entity_substr_list,
+                entity_ids_list,
+                conf_list,
+                entity_id_tags_list,
+            ):
+                entity_info = {}
+                is_abstract = rel.lower().replace("_", " ") in abstract_rels and not any(
+                    [f" {word} {entity_substr}" in context for word in ["the", "my", "his", "her"]]
+                )
+
+                f_entity_ids, f_confs, f_entity_id_tags = [], [], []
+                for entity_id, conf, entity_id_tag in zip(entity_ids, confs, entity_id_tags):
+                    if entity_id_tag.startswith("Abstract") and not is_abstract:
+                        pass
+                    else:
+                        f_entity_ids.append(entity_id)
+                        f_confs.append(conf)
+                        f_entity_id_tags.append(entity_id_tag)
+
+                if f_entity_ids:
+                    entity_info["entity_substr"] = entity_substr
+                    entity_info["entity_ids"] = f_entity_ids
+                    entity_info["confidences"] = [float(elem[2]) for elem in f_confs]
+                    entity_info["tokens_match_conf"] = [float(elem[0]) for elem in f_confs]
+                    entity_info["entity_id_tags"] = f_entity_id_tags
+                    entity_info_list.append(entity_info)
+            entity_info_batch.append(entity_info_list)
+    except Exception as e:
+        sentry_sdk.capture_exception(e)
+        logger.exception(e)
```

Comment:
@Ramimashkouk even in case of the error, length of entity_info_batch should be equal to length of the inpu data. So, you should assign default values here

Reply:
@dilyararimovna Done